docs/user-guide/deploying-application/image-labels-and-comments.md

```diff
@@ -0,0 +1,115 @@
+# Applying Labels and Comments
+
+## Introduction
+
+Typically in a CI pipeline, you [build container images](./triggering-ci.md), and the number of images gradually increases over a period of time. Devtron's image labels and comments feature helps you to mark and recall specific images from the repository by allowing you to add special instructions or notes to them. 
+
+For example:
+* You can label an image as `non-prod` to indicate that it is meant for 'Dev' or 'QA' environments, but not for production.
+* Add `hotfix image only` label to indicate a one-time patch on production.
+* Comments like `This image is buggy and shouldn't be used for deployment` to caution other users from deploying an unwanted image.
+
+![Figure 1: Labels and Comments](https://devtron-public-asset.s3.us-east-2.amazonaws.com/images/deploying-application/tag-comment/tag-and-comment.jpg)
+
+Such labels and comments will be visible only within Devtron, and will not propagate to your [container registry](../../reference/glossary.md#containeroci-registry) (say Docker Hub), unlike custom [image tag pattern](../creating-application/workflow/ci-pipeline.md#custom-image-tag-pattern). You may use it to simplify the management and [selection of container images](./triggering-cd.md#deploying-approved-image) for deployment.
+
+{% hint style="warning" %}
+Tagging labels and comments are supported only for images in workflows with at least one production deployment pipeline. In Devtron, you can go to **Global Configurations** → **Clusters & Environments** to identify a production environment by checking the 'Prod' label.
+{% endhint %}
+
+---
+
+## Adding Labels & Comments
+
+{% hint style="warning" %}
+### Who Can Perform This Action?
+Users need to have [Build & deploy permission](../global-configurations/authorization/user-access.md#role-based-access-levels) or above (along with access to the environment and application) to add labels and comments.
+{% endhint %}
+
+You can add labels and comments from the following pages:
+
+* [From Build & Deploy](#from-build--deploy)
+* [From Build History](#from-build-history)
+* [From Deployment History](#from-deployment-history) (only after deployment)
+* [From App Details](#from-app-details) (only after deployment)
+
+{% hint style="warning" %}
+You can add multiple labels to an image. Once added, they cannot be used for another image. <br />
```

Comment:
Cannot be used for another image in the application. The label can still be applied to an image in another application.